backend/src/main/java/trn/logistics/knapsack/service/KnapsackSolutionService.java

```diff
@@ -0,0 +1,48 @@
+package trn.logistics.knapsack.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import trn.logistics.knapsack.algorithm.TheAlgorithm;
+import trn.logistics.knapsack.database.KnapsackSolutionRepository;
+import trn.logistics.knapsack.database.MaterialRepository;
+import trn.logistics.knapsack.database.VehicleRepository;
+import trn.logistics.knapsack.dto.AlgorithmPair;
+import trn.logistics.knapsack.dto.KnapsackSolution;
+import trn.logistics.knapsack.dto.Material;
+import trn.logistics.knapsack.dto.Vehicle;
+
+import java.util.List;
+
+@Service
+public class KnapsackSolutionService {
+    @Autowired
+    private KnapsackSolutionRepository knapsackSolutionRepository;
+    @Autowired
+    private VehicleRepository vehicleRepository;
+    @Autowired
+    private MaterialRepository materialRepository;
+    @Autowired
+    private TheAlgorithm algorithm;
+
+    public KnapsackSolution createKnapsackSolutions(List<Long> vehicleIds, List<Long> materialIds, String name) {
+
+        List<Vehicle> vehicles = vehicleRepository.findAllById(vehicleIds);
+        List<Material> materials = materialRepository.findAllById(materialIds);
+
+        KnapsackSolution ks = new KnapsackSolution();
+        AlgorithmPair pair = algorithm.knapsackDistribution(materials, vehicles);
+        ks.setSelectedVehicles(pair.getVehiclelist());
+        ks.setNotLoadedMaterials(pair.getMateriallist());
+        ks.setName(name);
```

Comment:
Would look cleaner in a different order and if the `KnapsackSolution` would have a builder or an all args constructor.
```suggestion
        AlgorithmPair pair = algorithm.knapsackDistribution(materials, vehicles);

        KnapsackSolution ks = new KnapsackSolution(name, pair.getVehicleList(), pair.getMaterialList());
```

Or
```suggestion
        AlgorithmPair pair = algorithm.knapsackDistribution(materials, vehicles);

        KnapsackSolution ks = KnapsackSolution.builder()
                                              .selectedVehicles(pair.getVehiclelist())
                                              .notLoadedMaterials(pair.getMateriallist())
                                              .name(name)
                                              .build();
```